### ISSUE TITLE
Enhancement: input-date-picker sates

### ISSUE THREAD

**Issue by joeyHarig** [Esri/calcite-components#1947]
### Description
Add `disabled` and `readonly` states to the `calcite-input-date-picker` component. 

### Acceptance Criteria 
- `disabled` should disable the input and any pointer events
- `readonly` should set `readonly` on the input to true and also disable pointer events on the input

#### Which Component
- `calcite-input-date-picker`

#### Example Use Case
Disabled: When the date-input within a form is waiting on other inputs and is not yet enabled
Readonly: When the date has already been set and can not be changed.

**Comment by driskull** [comment]
@paulcpederson @eriklharper @macandcheese We don't have `readonly` on `calcite-input` so we would need to add that first, correct? Is there a reason we don't have it currently?

**Comment by paulcpederson** [comment]
No, we should probably have `readonly` for inputs to maintain API parity with native HTML.

**Comment by macandcheese** [comment]
Yeah agree.. right now you can add readonly and the styles will correctly apply, and the attribute will render on the child input element... however it’s not defined as a custom prop - it’s just getting spread if present - so it’s not getting documented.

Should add that so it’s on the calcite input and to the point of the issue probably anything that wraps one like a date, or time input.

**Comment by eriklharper** [comment]
I'm in favor of removing spread attributes as a general principle.  Best to always explicitly define the attributes being used.

**Comment by driskull** [comment]
> I'm in favor of removing spread attributes as a general principle. Best to always explicitly define the attributes being used and supported.

Yeah, we shouldn't be passing everything into the internals of the component.

**Comment by jcfranco** [comment]
The disabled prop support was handled by #2655, so I'll repurpose this issue to add read-only support to the input date/time pickers. cc @benelan

**Comment by jcfranco** [comment]
@benelan can we squeeze this one for this release?

**Comment by benelan** [comment]
> @benelan can we squeeze this one for this release?

Sure, triaged!

**Comment by jcfranco** [comment]
@Elijbet Can you look at this one for this release?

**Comment by Elijbet** [comment]
> @Elijbet Can you look at this one for this release?

Sure thing.
